dev/docs

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+set -eou pipefail
+
+cargo doc --no-deps --workspace --document-private-items --open 
```

Comment:
@neekolas Open to suggestions for what flag option(s) you want to have here when building the docs.

Reply:
Can we drop this script in favour of the one in https://github.com/xmtp/libxmtp/pull/456, which is more optimized for GH pages?